snapshots/devmapper/pool_device.go

```diff
@@ -207,25 +206,23 @@ func (p *PoolDevice) transition(ctx context.Context, deviceName string, tryingSt
 	})
 
 	if uerr != nil {
-		result = multierror.Append(result, uerr)
+		result = append(result, uerr)
 	}
 
-	return unwrapError(result)
+	return unwrapError(errors.Join(result...))
 }
 
-// unwrapError converts multierror.Error to the original error when it is possible.
-// multierror 1.1.0 has the similar function named Unwrap, but it requires Go 1.14.
-func unwrapError(e *multierror.Error) error {
+func unwrapError(e error) error {
 	if e == nil {
 		return nil
 	}
 
-	// If the error can be expressed without multierror, return the original error.
-	if len(e.Errors) == 1 {
-		return e.Errors[0]
+	if joinErr, ok := e.(interface{ Unwrap() []error }); ok {
+		if errs := joinErr.Unwrap(); len(errs) == 1 {
+			return errs[0]
```

Comment:
If I recall this right, we required this unwrapping helper for error comparisons to work. Which can now be replaced with `error.Is`. I wonder if we can now remove this?

Reply:
thanks for the context. I remember there're 2 places where the unwrap is used. Let me take a look at both.

Reply:
Digging into a little bit, `unwrapError` was introduced in #4437. At that time it seems `multierror 1.0.0` is used (or at least older than 1.1.0).

My guess is `unwrapError` was required to expose the underlying `unix.ENODATA` error within `multierror` which was handled below

https://github.com/containerd/containerd/blob/fb786b2f17b2c7d7ea5d9c9009d980bc03379952/snapshots/devmapper/pool_device.go#L545-L545

It was needed because at that time, `errors.Is` seems not work with `multierror`:

In this playground with `multierror 1.0.0`, `errors.Is` returns `false`: https://go.dev/play/p/29rAHWAV310
In this playground with `multierror 1.1.0`, `errors.Is` returns `true`: https://go.dev/play/p/UpFMxRkmjcc

And now since `errors.Is` works with both `multierror 1.1.0` and `errors.Join`, we can safely remove this regardless :)